File: e2e/php/tests/AuthTest.php
<?php

declare(strict_types=1);

namespace Kreuzberg\E2e;

use PHPUnit\Framework\TestCase;
use Kreuzcrawl\Kreuzcrawl;

/** E2e tests for category: auth. */
final class AuthTest extends TestCase
{
    /** Sends HTTP Basic authentication header */
    public function test_auth_basic_http(): void
    {
        $engine = Kreuzcrawl::createEngine(null);
        $url = getenv('MOCK_SERVER_URL') . '/fixtures/auth_basic_http';
        $result = Kreuzcrawl::scrape($engine, $url);
        $this->assertEquals(true, $result->auth_header_sent);
        $this->assertEquals(200, $result->status_code);
    }

    /** Sends Bearer token in Authorization header */
    public function test_auth_bearer_token(): void
    {
        $engine = Kreuzcrawl::createEngine(null);
        $url = getenv('MOCK_SERVER_URL') . '/fixtures/auth_bearer_token';
        $result = Kreuzcrawl::scrape($engine, $url);
        $this->assertEquals(true, $result->auth_header_sent);
        $this->assertEquals(200, $result->status_code);
    }

    /** Sends authentication via custom header (X-API-Key) */
    public function test_auth_custom_header(): void
    {
        $engine = Kreuzcrawl::createEngine(null);
        $url = getenv('MOCK_SERVER_URL') . '/fixtures/auth_custom_header';
        $result = Kreuzcrawl::scrape($engine, $url);
        $this->assertEquals(true, $result->auth_header_sent);
        $this->assertEquals(200, $result->status_code);
    }
}

File: e2e/rust/tests/auth_test.rs
//! E2e tests for category: auth

use kreuzcrawl::scrape;

#[test]
fn test_auth_basic_http() {
    // Sends HTTP Basic authentication header
    let engine = None;
    let url = None;
    let result = scrape(engine, url).expect("should succeed");
    assert_eq!(result.auth_header_sent, "true", "equals assertion failed");
    assert_eq!(result.status_code, "200", "equals assertion failed");
}

#[test]
fn test_auth_bearer_token() {
    // Sends Bearer token in Authorization header
    let engine = None;
    let url = None;
    let result = scrape(engine, url).expect("should succeed");
    assert_eq!(result.auth_header_sent, "true", "equals assertion failed");
    assert_eq!(result.status_code, "200", "equals assertion failed");
}

#[test]
fn test_auth_custom_header() {
    // Sends authentication via custom header (X-API-Key)
    let engine = None;
    let url = None;
    let result = scrape(engine, url).expect("should succeed");
    assert_eq!(result.auth_header_sent, "true", "equals assertion failed");
    assert_eq!(result.status_code, "200", "equals assertion failed");
}

